src/liballoc/tests/btree/map.rs

```diff
@@ -309,6 +309,41 @@ fn test_iter_mixed() {
     test(size, map.into_iter());
 }
 
+#[test]
+fn test_iter_min_max() {
+    let mut a = BTreeMap::new();
+    assert!(a.iter().min().is_none());
+    assert!(a.iter().max().is_none());
+    assert!(a.iter_mut().min().is_none());
+    assert!(a.iter_mut().max().is_none());
+    assert!(a.range(..).min().is_none());
+    assert!(a.range(..).max().is_none());
+    assert!(a.range_mut(..).min().is_none());
+    assert!(a.range_mut(..).max().is_none());
+    assert!(a.keys().min().is_none());
+    assert!(a.keys().max().is_none());
+    assert!(a.values().min().is_none());
+    assert!(a.values().max().is_none());
+    assert!(a.values_mut().min().is_none());
+    assert!(a.values_mut().max().is_none());
```

Comment:
Why did I not write `assert_eq!(..., None)`? Some invisible force told me to, possibly aliens, possibly the unit test framework of some other language that rejects a typeless None as ambiguous.